OpenRA.Mods.Common/Effects/RallyPointIndicator.cs

```diff
@@ -44,6 +48,24 @@ public void Tick(World world)
 			if (cachedLocation != rp.Location)
 			{
 				cachedLocation = rp.Location;
+
+				var rallyPos = world.Map.CenterOfCell(cachedLocation);
+				var exitPos = building.CenterPosition;
+
+				// Find closest exit
+				var dist = int.MaxValue;
+				foreach (var exit in exits)
+				{
+					var ep = building.CenterPosition + exit.SpawnOffset;
+					var len = (rallyPos - ep).Length;
+					if (len < dist)
+					{
+						dist = len;
+						exitPos = ep;
+					}
+				}
+
+				targetLine = new[] { exitPos, rallyPos };
```

Comment:
Please initialize this (`WPos[] targetLine = new WPos[2];`) and just update the existing array rather than creating a new one each time.